File: Day242_2025/Q242_Leetcode.py
from typing import List


class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        row = [[False] * 9 for _ in range(9)]
        col = [[False] * 9 for _ in range(9)]
        grid = [[False] * 9 for _ in range(9)]
        
        for i in range(9):
            for j in range(9):
                if board[i][j] == '.':
                    continue
                num = int(board[i][j]) - 1
                k = 3 * (i // 3) + (j // 3)
                
                if row[i][num] or col[j][num] or grid[k][num]:
                    return False
                
                row[i][num] = col[j][num] = grid[k][num] = True
        return True

File: Day242_2025/Q242_Leetcode.java
class Q242_Leetcode {
    public boolean isValidSudoku(char[][] board) {
        boolean[][] row = new boolean[9][9];
        boolean[][] col = new boolean[9][9];
        boolean[][] grid = new boolean[9][9];
        
        for (int i = 0; i < 9; i++) {
            for (int j = 0; j < 9; j++) {
                if (board[i][j] == '.') continue;
                
                int num = board[i][j] - '1';
                int k = 3 * (i / 3) + (j / 3);
                
                if (row[i][num] || col[j][num] || grid[k][num]) {
                    return false;
                }
                
                row[i][num] = col[j][num] = grid[k][num] = true;
            }
        }
        return true;
    }
}
